src/System.Collections/src/System/Collections/Generic/HashSet.cs

```diff
@@ -1731,14 +1955,30 @@ private static bool AreEqualityComparersEqual(HashSet<T> set1, HashSet<T> set2)
         /// Workaround Comparers that throw ArgumentNullException for GetHashCode(null).
         /// </summary>
         /// <param name="item"></param>
+        /// <param name="comparer"></param>
         /// <returns>hash code</returns>
-        private int InternalGetHashCode(T item)
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static int InternalGetHashCode(T item, IEqualityComparer<T>? comparer)
         {
             if (item == null)
             {
                 return 0;
             }
-            return _comparer.GetHashCode(item) & Lower31BitMask;
+
+            int hashCode = comparer?.GetHashCode(item) ?? item.GetHashCode();
+            return hashCode & Lower31BitMask;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private int InternalGetHashCode(int hashCode)
+        {
+            return hashCode & Lower31BitMask;
+        }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private bool InternalEquals(T x, T y, IEqualityComparer<T>? comparer)
```

Comment:
Wrappers like this add overhead for larger structs; even with AggressiveInlining.

Reply:
The fine-tuned Dictionary implementation avoids them for a good reason, so you may want to avoid them too - if you are shooting for HashSet to be fine-tuned in the same way.

Reply:
I see. Actually it's only used in `SymmetricExceptWith` now. All the other usages (in `Remove`, `TryGetValue` etc.) have been replace by the optimization. I'll manually inline the method.